src/uct/api/uct_def.h

```diff
@@ -235,4 +238,77 @@ typedef size_t (*uct_pack_callback_t)(void *dest, void *arg);
 typedef void (*uct_unpack_callback_t)(void *arg, const void *data, size_t length);
 
 
+/**
+ * @ingroup UCT_TAG
+ * @brief Callback to process unexpected eager tagged message.
+ *
+ * This callback is invoked when tagged message sent by eager protocol is
+ * arrived and there is no corresponding tag posted.
+ *
+ * @note The callback is always invoked from the context (thread, process)
+ *       that called @a uct_iface_progress().
+ *
+ * @note It is allowed to call other communication routines from the callback.
+ *
+ * @param [in]  arg     User-defined argument
+ * @param [in]  data    Points to the received unexpected data.
+ * @param [in]  length  Length of data.
+ * @param [in]  desc    Points to the received descriptor, at the beginning of
+ *                      the user-defined rx_headroom.
+ * @param [in]  stag    Tag from sender.
+ * @param [in]  imm     Immediate data from sender.
+ *
+ * @warning If the user became the owner of the @a desc (by returning
+ *          @ref UCS_INPROGRESS) the descriptor must be released later by
+ *          @ref uct_iface_release_desc by the user.
+ *
+ * @retval UCS_OK         - descriptor was consumed, and can be released
+ *                          by the caller.
+ * @retval UCS_INPROGRESS - descriptor is owned by the callee, and would be
+ *                          released later.
+ */
+typedef ucs_status_t (*uct_tag_unexp_eager_cb_t)(void *arg, void *data,
+                                                 size_t length, void *desc,
+                                                 uct_tag_t stag,  uint64_t imm);
+
+
+/**
+ * @ingroup UCT_TAG
+ * @brief Callback to process unexpected rendezvous tagged message.
+ *
+ * This callback is invoked when rendezvous send notification is arrived
+ * and there is no corresponding tag posted.
```

Comment:
Minor grammar cleanup:
  This callback is invoked when a rendezvous send notification has
  arrived and no corresponding tag has been posted.